x-pack/plugins/security_solution/public/timelines/components/timeline/header/save_timeline_button.test.tsx

```diff
@@ -0,0 +1,125 @@
+jest.mock('../../../../common/components/user_privileges');
+jest.mock('../../flyout/header/selectors', () => {
+  return {
+    getTimelineStatusByIdSelector: jest.fn().mockReturnValue(() => ({
+      status: 'draft',
+      isSaving: false,
+    })),
+  };
+});
+
+const props: SaveTimelineButtonProps = {
+  timelineId: 'timeline-1',
+};
+
+const TestSaveTimelineButton = (_props: SaveTimelineButtonProps) => (
+  <TestProviders>
+    <SaveTimelineButton {..._props} />
+  </TestProviders>
+);
+
+jest.mock('raf', () => {
+  return jest.fn().mockImplementation((cb) => cb());
+});
+
+describe('SaveTimelineButton', () => {
+  it('should disable the save timeline button when the user does not have write acceess', () => {
+    (useUserPrivileges as jest.Mock).mockReturnValue({
+      kibanaSecuritySolutionsPrivileges: { crud: false },
+    });
+    render(
+      <TestProviders>
+        <SaveTimelineButton {...props} />
+      </TestProviders>
+    );
+    expect(screen.getByRole('button')).toBeDisabled();
+  });
+
+  describe('with draft timeline', () => {
+    it('should not show the save modal if user does not have write access', async () => {
+      (useUserPrivileges as jest.Mock).mockReturnValue({
+        kibanaSecuritySolutionsPrivileges: { crud: false },
+      });
+      render(<TestSaveTimelineButton {...props} />);
+
+      expect(screen.queryByTestId(TEST_ID.SAVE_TIMELINE_MODAL)).not.toBeInTheDocument();
+
+      const saveTimelineBtn = screen.getByRole('button');
+
+      fireEvent.click(saveTimelineBtn);
+
+      await waitFor(() => {
+        expect(screen.queryAllByTestId(TEST_ID.SAVE_TIMELINE_MODAL)).toHaveLength(0);
+      });
+    });
+
+    it('should show the save modal when user has crud privileges', async () => {
+      (useUserPrivileges as jest.Mock).mockReturnValue({
+        kibanaSecuritySolutionsPrivileges: { crud: true },
+      });
+      render(<TestSaveTimelineButton {...props} />);
+      expect(screen.queryByTestId(TEST_ID.SAVE_TIMELINE_MODAL)).not.toBeInTheDocument();
+
+      const saveTimelineBtn = screen.getByRole('button');
+
+      fireEvent.click(saveTimelineBtn);
+
+      await waitFor(() => {
+        expect(screen.getByTestId(TEST_ID.SAVE_TIMELINE_MODAL)).toBeVisible();
+      });
+    });
+  });
+
+  describe('with active timeline', () => {
+    beforeAll(() => {
+      (getTimelineStatusByIdSelector as jest.Mock).mockReturnValue(() => ({
+        status: TimelineStatus.active,
+        isSaving: false,
+      }));
+    });
+
+    it('should open the timeline save modal', async () => {
+      (useUserPrivileges as jest.Mock).mockReturnValue({
+        kibanaSecuritySolutionsPrivileges: { crud: true },
+      });
+      render(<TestSaveTimelineButton {...props} />);
+
+      const saveTimelineBtn = screen.getByRole('button');
+
+      fireEvent.click(saveTimelineBtn);
+
+      await waitFor(() => {
+        expect(screen.getByTestId(TEST_ID.SAVE_TIMELINE_MODAL)).toBeInTheDocument();
```

Comment:
Good point, will add that!

Reply:
@logeekal The color change and the tooltip behaviour is not observable from jest since tooltips don't work there properly. I added a Cypress test instead: https://github.com/elastic/kibana/pull/169239/commits/22b3c7050fa1bca1a00a5041313215d545f7155b 🎉

Reply:
Okay.. Cool. Thank you.